### DISCUSSION TITLE
#include <type_traits>

### DISCUSSION THREAD

**Discussion by robbie-sps** [andrew-gresyk/HFSM2#87] [General]
At commit 04a579531926c3ade7c950f366f0665b00b81c25 there is a comment that use of std::aligned_storage<> was removed.

In machine.hpp, there is the following include:
```
#include <type_traits>		// std::aligned_storage<>
```
Is it safe to assume that this include can be removed? I tried it and I didn't get any compilation errors.

I am having compilation errors with building hfsm2 for a microcontroller target running NuttX, which doesn't have all the std lib, including typeindex (which I disabled with HFSM2_DISABLE_TYPEINDEX) and type_traits.

Thanks :)

**Comment by andrew-gresyk** [comment]
Good catch, thanks for reporting.

`#include <type_traits>` has been removed in version [2.3.2](https://github.com/andrew-gresyk/HFSM2/releases/tag/2.3.2).
